go/tools/builders/link.go

```diff
@@ -110,33 +109,34 @@ func link(args []string) error {
 	parseXdef := func(xdef string) (pkg, name, value string, err error) {
 		eq := strings.IndexByte(xdef, '=')
 		if eq < 0 {
-			return "", "", "", fmt.Errorf("-X or -Xstamp flag does not contain '=': %s", xdef)
+			return "", "", "", fmt.Errorf("-X flag does not contain '=': %s", xdef)
 		}
 		dot := strings.LastIndexByte(xdef[:eq], '.')
 		if dot < 0 {
-			return "", "", "", fmt.Errorf("-X or -Xstamp flag does not contain '.': %s", xdef)
+			return "", "", "", fmt.Errorf("-X flag does not contain '.': %s", xdef)
 		}
 		pkg, name, value = xdef[:dot], xdef[dot+1:eq], xdef[eq+1:]
 		if pkg == *packagePath {
 			pkg = "main"
 		}
 		return pkg, name, value, nil
 	}
-	for _, xdef := range xstamps {
-		pkg, name, key, err := parseXdef(xdef)
-		if err != nil {
-			return err
-		}
-		if value, ok := stampMap[key]; ok {
-			goargs = append(goargs, "-X", fmt.Sprintf("%s.%s=%s", pkg, name, value))
-		}
-	}
 	for _, xdef := range xdefs {
 		pkg, name, value, err := parseXdef(xdef)
 		if err != nil {
 			return err
 		}
-		goargs = append(goargs, "-X", fmt.Sprintf("%s.%s=%s", pkg, name, value))
+		var missingKey bool
+		value = regexp.MustCompile("\\{.*?\\}").ReplaceAllStringFunc(value, func(key string) string {
```

Comment:
Why do you need the `?` following `.*`? `*` is the 0 or more operator so it feels like making that optional is unnecessary. I also think that we would end up in a index panic with this when someone passes in `{}` to be matched.

WDYT about doing something like:

```
		value = regexp.MustCompile(`\{.+\}`).ReplaceAllStringFunc(value, func(key string) string {Write<br class="Apple-interchange-newline">
```

2 changes in there

1. Uses raw string syntax
2. Uses the `+` operator over the `*` operator which would only match if there is one or more entry. That would mean there is always at least a single character inside the `{}` which would prevent the negative index panic I think I see.

Reply:
The question mark here put the `*` in [lazy mode](https://en.wikipedia.org/wiki/Regular_expression#Lazy_matching)

So for a string like `ab{c}d{e}f`, it produces two matches `{c}` and `{e}` instead of one match `{c}d{e}`.

* [x] Raw string
* [x] Use `+` intead of `*`